crates/bevy_ecs/src/system/combinator.rs

```diff
@@ -88,7 +88,7 @@ use super::{IntoSystem, ReadOnlySystem, System};
     label = "invalid system combination",
     note = "the inputs and outputs of `{A}` and `{B}` are not compatible with this combiner"
 )]
-pub trait Combine<A: System, B: System> {
+pub trait Combine<A: System<In = Self::In>, B: System<In = Self::In>> {
```

Comment:
Isn't this a breaking change? It will force the two systems to have the same input type, so e.g. piping would no longer be implementable this way (not that `bevy`'s implementation used `Combine`).